api/transport/request.go

```diff
@@ -31,6 +31,23 @@ import (
 
 // Request is the low level request representation.
 type Request struct {
+	// ID is a unique identifier for a request/response pair. This MAY be a
+	// trace ID or UUID.
+	//
+	// This MAY be set by transports or middleware.
+	ID string
```

Comment:
@nomis52 are we going to be needing more of these? now's the time to ask for more if you foresee any other obvious ones.